src/main/java/org/veupathdb/eda/binaryfiles/dumper/MaxIdLengthFinder.java

```diff
@@ -0,0 +1,40 @@
+package org.veupathdb.eda.binaryfiles.dumper;
+
+import org.veupathdb.service.eda.ss.model.tabular.TabularResponses;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * ResultConsumer that tracks the longest ID. Returns the max length once the entire result has been consumed.
+ */
+public class MaxIdLengthFinder implements TabularResponses.ResultConsumer {
+  private static final int INDEX_OF_ID = 0;
+  private int maxLength = 0;
+  private boolean done = false;
+  private boolean firstRow = true;
+
+  @Override
+  public void consumeRow(List<String> record) throws IOException {
+    if (firstRow) { firstRow = false; return; } // skip header
+
+    int numBytes = record.get(INDEX_OF_ID).getBytes(StandardCharsets.UTF_8).length + 4;
```

Comment:
Add a comment here about why you add 4 at the end.

Reply:
Added, also changed the 4 to Integer.BYTES for clarity